pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -98,30 +122,63 @@ func (rc *remoteClient) setConfig(watchCtx context.Context, kubeconfig []byte) e
 	}
 	rc.client = remoteClient
 
-	newWatcher, err := rc.client.Watch(watchCtx, &kueue.WorkloadList{})
+	err = rc.startWatcher(watchCtx, kueue.GroupVersion.WithKind("Workload").GroupKind().String(), &workloadKueueWatcher{})
 	if err != nil {
-		return nil
+		return err
+	}
+
+	// add a watch for all the adapters implementing multiKueueWatcher
+	for kind, adapter := range adapters {
+		watcher, implementsWatcher := adapter.(multiKueueWatcher)
+		if !implementsWatcher {
+			continue
+		}
+		err := rc.startWatcher(watchCtx, kind, watcher, client.HasLabels{constants.PrebuiltWorkloadLabel})
+		if err != nil {
+			// not being able to setup a watcher is not ideal but we can function
```

Comment:
I would say this is a hard error, to fail fast. Otherwise it may seem like working but the important feature of live updates will be discovered as broken late.